### PR TITLE
SDK-532 [Node.js SDK] Proxy support

### PR DESCRIPTION
**Pull request by artemgolozubov**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-532](https://incountry.atlassian.net/browse/SDK-532)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [x] Please put the target release vehicle this needs to be bundled with.
- [x] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

**Review by RommeroInCountry — DISMISSED** [review]

File: tests/integration/proxy-server-support-tests.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+    const target = fixRequestTarget(req);
+    fixRequestBodyAudience(req);
+
+    req.bodyPacked = packResponseBody(req.body, contentType);
+    proxy.web(req, res, getProxyRequestBaseOptions(target, req));
+  });
+
+  server = http.createServer(proxyApp);
+  server.listen(proxyPort, proxyHost);
```
Comment (igor-alekseev-dev) [inline tests/integration/proxy-server-support-tests.ts:112]: Maybe it's better to return `server` instance here

**Review by igor-alekseev-dev — APPROVED** [review]

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]
Please provide README updates for this feature

File: README.md [hunk 2]
```diff
@@ -120,6 +121,15 @@ type StorageOptions = {
     maxBodyLength?: NonNegativeInt; // Maximum allowed HTTP request body length, in bytes; Defaults to 100 Mb
     retryBaseDelay?: NonNegativeInt; // Base delay for HTTP request retries in case of server high load, in milliseconds; Defaults to 1s
     retryMaxDelay?: NonNegativeInt; // Maximum delay for HTTP request retries in case of server high load, in milliseconds; Defaults to 32s
+    proxy?: { // Proxy server configuration
+      host: string;
+      port: number;
+      auth?: {
+        username: string;
+        password: string;
+      };
```
Comment (RommeroInCountry) [inline README.md:130]: Typo, should be ,

File: README.md [hunk 3]
```diff
@@ -310,6 +320,34 @@ await storage.validate();
 
 The `validate` method fetches the secret using `getSecrets` and validates it. If custom encryption configurations were provided they would also be checked with all the matching secrets.
 
+#### Proxy server support
+
+The SDK allows to use proxy server for all requests.
```
Comment (RommeroInCountry) [inline README.md:325]: for all HTTP requests

File: README.md [hunk 4]
```diff
@@ -310,6 +320,34 @@ await storage.validate();
...
+
+Below you can find the example of how to specify your proxy server configuration:
```
Comment (RommeroInCountry) [inline README.md:327]: an example

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -310,6 +320,34 @@ await storage.validate();
...
+    clientSecret: '<client_secret>',
+  },
+  httpOptions: {
+    proxy: {
+      host: '127.0.0.1',
+      port: 10174,
+      auth: {
+        username: '<username>',
+        password: '<password>',
+      };
```
Comment (RommeroInCountry) [inline README.md:344]: Typo, should be ,

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

**Review by RommeroInCountry — APPROVED** [review]

**Review by igor-alekseev-dev — APPROVED** [review]

**Pull request merged**